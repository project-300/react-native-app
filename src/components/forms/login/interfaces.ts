import { TextStyle, ViewStyle } from 'react-native';
import { CommonProps } from '../../../types/common';

export interface Styles {
	container: ViewStyle;
	input: TextStyle;
	button: ViewStyle;
	buttonText: TextStyle;
	signUpLink: TextStyle;
	showPasswordIconContainer: ViewStyle;
	showPasswordIcon: TextStyle;
	uiButton: ViewStyle;
	textInput: ViewStyle;
	closeButton: ViewStyle;
}

export interface Props extends CommonProps {
	keyboardOpen: boolean;
	isLoggingIn: boolean;
	isLoggedIn: boolean;
	login(u: string, p: string): Promise<boolean>;
}

export interface State {
	username: string;
	password: string;
	hidePassword: boolean;
}
