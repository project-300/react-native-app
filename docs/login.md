## Login Screen

#### State Mapping
| State | Source |
|-----|-----|
| username | Username field input |
| password | Password field input |
| error | Manual validation checks or AWS Incognito validation checks |

#### Submission
Upon clicking the submit button, a call is made to AWS Amlify. The username and password is sent via the Amlify SDK to be validated.

The account details are validated with AWS Cognito.

#### Failure
If the user is missing either their username or password, they will be alerted before making the call to AWS.

If the validation is unsuccessful, the appropriate error message that was returned from AWS Cognito is shown to the user. 

#### Success 
If the validation is successful, the details returned from the API call is then sent to AWS Lambda to be logged in our database. 
The app then navigates to the home screen.
