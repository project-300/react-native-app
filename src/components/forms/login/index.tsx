import React, { Component, ReactElement } from 'react';
import {
	Text,
	View,
	TextInput,
	TouchableOpacity, Platform
} from 'react-native';
import { State, Props } from './interfaces';
import toastr from '../../../helpers/toastr';
import formStyles from '../../../styles/forms';

export class LoginForm extends Component<Props, State> {

	public constructor(props: Props) {
		super(props);

		this.state = {
			email: '',
			password: '',
			hidePassword: true
		};
	}

	private _loginAttempt = async (): Promise<void> => {
		this.setState({ hidePassword: true });

		const { email, password } = this.state;

		if (!email) return toastr.error('Username is missing');
		if (!password) return toastr.error('Password is missing');

		const res: boolean = await this.props.login(email, password);
		res && this.props.navigation.navigate('Home');
	}

	public render(): ReactElement {
		/*
		 	Add bottom margin to form when on Android and keyboard open.
		 	The sign up button is hidden behind keyboard otherwise.
		*/

		return (
			<View style={ { marginBottom: Platform.OS === 'android' && this.props.keyboardOpen ? 80 : 0 } }>
				<TextInput
					placeholder='EMAIL ADDRESS'
					placeholderTextColor='black'
					onChangeText={ (email: string): void => this.setState({ email }) }
					autoCorrect={ false }
					autoCompleteType={ 'off' }
					autoCapitalize='none'
					keyboardType={ 'email-address' }
					style={ formStyles.input } />

				<TextInput
					placeholder='PASSWORD'
					placeholderTextColor='black'
					secureTextEntry={ true }
					autoCapitalize='none'
					onChangeText={ (password: string): void => this.setState({ password })}
					style={ formStyles.input } />

				<TouchableOpacity
					style={ formStyles.largeButton }
					onPress={ this._loginAttempt }
					disabled={ this.props.isLoggingIn || !this.state.email || !this.state.password }>
					<Text style={ formStyles.buttonText }>
						SIGN IN
					</Text>
				</TouchableOpacity>
			</View>
		);
	}
}
