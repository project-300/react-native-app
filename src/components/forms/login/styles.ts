import { Dimensions, StyleSheet } from 'react-native';
import { Styles } from './interfaces';

const { width } = Dimensions.get('window');

const styles: Styles = StyleSheet.create<Styles>({
	container: {
		width: width * 0.8,
		alignSelf: 'center'
	}
});

export default styles;
