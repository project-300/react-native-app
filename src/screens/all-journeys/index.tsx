import React, { Component, ReactElement } from 'react';
import {
  ScrollView,
  FlatList,
  Text,
  View,
  TouchableOpacity
} from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';
import { Props, State } from './interfaces';
import { AppState } from '../../store';
import { AllJourneysListState } from '../../types/redux-reducer-state-types';
import { Journey } from '@project-300/common-types';
import { getAllJourneys, updateAddUserJourney } from '../../redux/actions';
import {
  Container,
  Tab,
  Tabs,
  Content,
  Card,
  CardItem,
  Body
} from 'native-base';
import Spinner from 'react-native-loading-spinner-overlay';

class AllJourneys extends Component<Props, State> {
  public constructor(props: Props) {
    super(props);
  }

  public async componentDidMount(): Promise<void> {
    await this.props.getAllJourneys();
  }

  private async _joinJourney(journey: Journey): Promise<void> {
    await this.props.updateAddUserJourney(journey);
  }

  private _renderRow = ({
    item,
    index
  }: {
    item: Journey;
    index: number;
  }): ReactElement => {
    const journey: Journey = item;
    return (
      <Card>
        <CardItem header bordered style={styles.cardHeader}>
          <Text style={styles.journeyHeading}>
            <Text style={styles.bold}>{journey.origin.name}</Text> to{' '}
            <Text style={styles.bold}>{journey.destination.name}</Text>
          </Text>
        </CardItem>
        <CardItem bordered>
          <Body>
            <Text>The journey begins at {journey.times.leavingAt}</Text>
            <Text>
              {journey.seatsLeft} / {journey.totalNoOfSeats} seats left
            </Text>
            <Text>{journey.pricePerSeat} euro per seat</Text>
          </Body>
        </CardItem>
        <CardItem footer bordered>
          {/* <TouchableOpacity
            style={styles.button}
            onPress={() => this._joinJourney(journey)}
          >
            <Text style={styles.buttonText}>Join</Text>
          </TouchableOpacity> */}
          <TouchableOpacity
            style={styles.button}
            onPress={(): void => {
              this.props.navigation.navigate('ViewJourney', journey);
            }}
          >
            <Text style={styles.buttonText}>View</Text>
          </TouchableOpacity>
        </CardItem>
      </Card>
    );
  };

  public render(): ReactElement {
    return (
      <Container>
        <Content>
          <Spinner visible={this.props.isFetching} />
          <ScrollView style={styles.container}>
            <FlatList
              data={this.props.journeys}
              renderItem={this._renderRow}
              keyExtractor={(item: Journey): string => item.journeyId}
            />
          </ScrollView>
        </Content>
      </Container>
    );
  }
}

const mapStateToProps = (state: AppState): AllJourneysListState => ({
  ...state.allJourneysReducer
});

export default connect(mapStateToProps, {
  getAllJourneys,
  updateAddUserJourney
})(AllJourneys);
