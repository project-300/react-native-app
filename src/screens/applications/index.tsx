import React, { Component, ReactElement } from 'react';
import {
	ScrollView,
	FlatList,
	Text,
	View,
	TouchableOpacity
} from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';
import { Props, State } from './interfaces';
import WS from '../../api/websocket';
import { ADMIN_DRIVER_APPLICATIONS } from '../../constants/websocket-subscriptions';
import { storeApplications, approveApplication, deleteApplication } from '../../redux/actions';
import { DriverApplicationObject } from '@project-300/common-types';
import { AppState } from '../../store';
import { AdminDriverApplicationsState } from '../../types/redux-reducer-state-types';

class Applications extends Component<Props, State> {

	public constructor(props: Props) {
		super(props);
	}

	public async componentDidMount(): Promise<void> {
		await WS.subscribe(ADMIN_DRIVER_APPLICATIONS);
	}

	public async componentWillUnmount(): Promise<void> {
		await WS.unsubscribe(ADMIN_DRIVER_APPLICATIONS);
	}

	private _approve = async (userId: string): Promise<void> => {
		await this.props.approveApplication(userId);
	}

	private _delete = async (userId: string): Promise<void> => {
		await this.props.deleteApplication(userId);
	}

	private _renderRow = ({ item, index }: { item: DriverApplicationObject; index: number }): ReactElement => {
		const approvalButton =
			<TouchableOpacity
				style={ styles.button }
				onPress={ (): Promise<void> => this._approve(item.userId) }
			>
				<Text style={ styles.buttonText }>Approve</Text>
			</TouchableOpacity>;

		return (
			<View
				style={ styles.applicationRow }>
				<Text style={ styles.title }>
					Application #{ index + 1 }
				</Text>
				<Text>
					User ID: { item.userId }
				</Text>
				<Text>
					Approved: { item.approved ? 'Yes ---' : 'No' }
					{ item.times.approved ? ` at ${item.times.approved}` : '' }
				</Text>
				{ approvalButton }
				<TouchableOpacity
					style={ styles.button }
					onPress={ (): Promise<void> => this._delete(item.userId) }
				>
					<Text style={ styles.buttonText }>Delete</Text>
				</TouchableOpacity>
			</View>
		);
	}

	public render(): ReactElement {
		return (
			<ScrollView style={ styles.container }>
				<Text>This is a temporary WebSockets demo screen.</Text>
				<Text>There are { this.props.applications.length } applications.</Text>

				<FlatList
					data={ this.props.applications }
					extraData={ this.props }
					renderItem={ this._renderRow }
					keyExtractor={ (item: DriverApplicationObject): string => item.userId }
				/>
			</ScrollView>
		);
	}
}

const mapStateToProps = (state: AppState): AdminDriverApplicationsState => ({
	...state.adminDriverApplicationsReducer
});

export default connect(mapStateToProps, {
	storeApplications,
	approveApplication,
	deleteApplication
})(Applications);
