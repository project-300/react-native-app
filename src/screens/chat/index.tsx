import React, { Component, ReactElement } from 'react';
import {
	Text,
	ScrollView,
	View,
	TextInput,
	Image, NativeScrollEvent, Dimensions, TouchableOpacity, KeyboardAvoidingView, Platform, Keyboard
} from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';
import { Props, State, AnimationStyles, AnimationValues } from './interfaces';
import { chatSubscribe, chatUnsubscribe, getChatMessages } from '../../redux/actions';
import { AppState } from '../../store';
import { ChatState } from '../../types/redux-reducer-state-types';
import { NavigationEvents } from 'react-navigation';
import { Message } from '@project-300/common-types';
import Icon from 'react-native-vector-icons/FontAwesome5';
import { MessageService } from '../../services/message';
import DatesTimes from '../../services/dates-times';
import Animated, { Easing } from 'react-native-reanimated';
import { interpolateAnimation } from '../../animations/animations';

const { width } = Dimensions.get('window');
const { timing } = Animated;

export class Chat extends Component<Props, State> {

	private _scrollView: ScrollView;

	private animatedValues: AnimationValues = {
		newMessages: new Animated.Value(0),
		loadOldMessage: new Animated.Value(0)
	};

	private animatedStyles: AnimationStyles = {
		newMessagesButtonOpacity: 0,
		loadingBarWidth: 0
	};

	public constructor(props: Props) {
		super(props);

		this.state = {
			messageText: '',
			scrolledOffBottom: false,
			newMessageCount: 0,
			lastContentHeight: 1,
			isLoading: true,
			allowOldMessages: false,
			showLoadingBar: false
		};

		this.animatedValues = {
			newMessages: new Animated.Value(0),
			loadOldMessage: new Animated.Value(0)
		};

		this.animatedStyles = {
			newMessagesButtonOpacity: interpolateAnimation(this.animatedValues.newMessages, [ 0, 1 ], [ 0, 1 ]),
			loadingBarWidth: interpolateAnimation(this.animatedValues.loadOldMessage, [ 0, 1 ], [ 0, width ])
		};
	}

	public componentDidUpdate = (p: Props, s: State): void => {
		const newMessageCount: number = this.props.messages.length - p.messages.length;

		if (this.state.isLoading) setTimeout(() => this._scrollView.scrollToEnd({ animated: false }));
		else if (newMessageCount && !this.state.scrolledOffBottom) setTimeout(() => this._scrollView.scrollToEnd());

		if (this.state.isLoading && this.props.messages.length && newMessageCount) this.setState({ isLoading: false });
		if (newMessageCount && this.state.scrolledOffBottom && this.props.lastContentType === 'NEW' && !this.props.isLastMessageByOwnUser)
			this._showNewMessagesFloatButton(newMessageCount);
		if (newMessageCount && this.state.scrolledOffBottom && this.props.lastContentType === 'NEW' && this.props.isLastMessageByOwnUser)
			setTimeout(() => this._scrollView.scrollToEnd());
	}

	private _mountScreen = async (): Promise<void> => { // Triggered when this screen renders (navigated to)
		// await this.props.chatSubscribe(this.props.otherUserId);
		await this.props.chatSubscribe('cfa18855-2eb3-479b-9120-6ca3bef23fda');

		Keyboard.addListener('keyboardWillShow', () => this._scrollView.scrollToEnd({ animated: false }));
		Keyboard.addListener('keyboardDidShow', () => this._scrollView.scrollToEnd({ animated: false }));
		Keyboard.addListener('keyboardWillHide', () => this._scrollView.scrollToEnd({ animated: false }));
		Keyboard.addListener('keyboardDidHide', () => this._scrollView.scrollToEnd({ animated: false }));

		this._scrollView.scrollToEnd({ animated: false });
	}

	// 00c35391-2b2e-4155-b90d-47d6d922a245
	// cfa18855-2eb3-479b-9120-6ca3bef23fda

	private _unmountScreen = async (): Promise<void> => { // Triggered when the screen is navigated away from
		await this.props.chatUnsubscribe('cfa18855-2eb3-479b-9120-6ca3bef23fda');
	}

	private _showNewMessagesFloatButton = (newMessageCount: number): void => {
		this.setState({ newMessageCount: this.state.newMessageCount + newMessageCount }, () => {
			timing(this.animatedValues.newMessages, {
				duration: 400,
				toValue: 1,
				easing: Easing.inOut(Easing.ease)
			}).start();
		});
	}

	private _sendMessage = async (): Promise<void> => {
		const messageText: string = this.state.messageText;
		if (!this.state.messageText) return;

		// this._scrollView.scrollToEnd({ animated: false });

		await MessageService.sendMessage({
			text: messageText,
			chatId: '5862cb22-3c10-4723-8472-9698eb83d8e3'
		}, 'cfa18855-2eb3-479b-9120-6ca3bef23fda');

		this.setState({ messageText: '' });
	}

	private _scrollChatEvent = async ({ layoutMeasurement, contentOffset, contentSize }: NativeScrollEvent): Promise<void> => {
		const scrolledOffBottom: boolean = !(layoutMeasurement.height + contentOffset.y >= contentSize.height - 20);

		this.setState({
			scrolledOffBottom,
			lastContentHeight: contentSize.height,
			allowOldMessages: true,
			newMessageCount: scrolledOffBottom ? this.state.newMessageCount : 0
		});

		if (contentOffset.y <= 10 && this.state.allowOldMessages) await this._loadOldMessages();
	}

	private _loadOldMessages = async (): Promise<void> => {
		if (!this.props.chat || this.props.requestingOldMessages || !this.props.lastEvaluatedKey) return; // SK for message LEK is createdAt

		await this.props.getChatMessages(this.props.chat.chatId, this.props.lastEvaluatedKey.sk);
	}

	private _scrollToBottom = (): void => {
		this._scrollView.scrollToEnd();

		timing(this.animatedValues.newMessages, {
			duration: 100,
			toValue: 0,
			easing: Easing.inOut(Easing.ease)
		}).start(() => this.setState({ newMessageCount: 0 }));
	}

	private _renderMessage = (message: Message): ReactElement => {
		return <View style={ [
			styles.messageContainer,
			message.userOwnMessage ? { alignSelf: 'flex-end' } : { }
		] } key={ message.sk }>
			{
				!message.userOwnMessage &&
					<Image style={ [ styles.avatar, styles.senderAvatar ] } source={ { uri: message.createdBy.avatar } } />
			}
			<View
				style={ [
					styles.messageBubble,
					message.userOwnMessage ? styles.userOwnMessageBubble : { }
				] }
			>
				<Text style={ [
					styles.messageText,
					message.userOwnMessage ? { } : styles.userOwnMessageText
				] }>{ message.text }</Text>
				<Text style={ [ styles.messageTime ] }>{ DatesTimes.dayAndTime(message.times.createdAt) }</Text>
			</View>
			{
				message.userOwnMessage &&
					<Image style={ [ styles.avatar, styles.ownAvatar ] } source={ { uri: message.createdBy.avatar } } />
			}
		</View>;
	}

	private _renderNavigationEvents = (): ReactElement =>
		<NavigationEvents
			onWillFocus={ this._mountScreen }
			onDidBlur={ this._unmountScreen }
		/>

	public render(): ReactElement {
		return (
			<KeyboardAvoidingView style={ { flex: 1, justifyContent: 'center' } } behavior={ Platform.OS === 'ios' ? 'height' : undefined }
				keyboardVerticalOffset={ 110 }
			>
				<ScrollView
					style={ styles.chatWindow }
					ref={ (ref: ScrollView): void => { this._scrollView = ref; }}
					onMomentumScrollEnd={
						({ nativeEvent }: { nativeEvent: NativeScrollEvent }): Promise<void> => this._scrollChatEvent(nativeEvent)
					}
					onScrollEndDrag={
						({ nativeEvent }: { nativeEvent: NativeScrollEvent }): Promise<void> => this._scrollChatEvent(nativeEvent)
					}
					onContentSizeChange={ (contentWidth: number, contentHeight: number): void => {
						if (!this.state.isLoading && this.props.lastContentType === 'OLD') {
							this._scrollView.scrollTo({ x: 0, y: contentHeight - this.state.lastContentHeight, animated: false });
							this._scrollView.scrollTo({ x: 0, y: contentHeight - this.state.lastContentHeight - 20, animated: true });
						}
					} }
				>
					{ this._renderNavigationEvents() }

					{
						!this.props.lastEvaluatedKey && this.props.messages.length > 10 &&
							<Text
								style={ {
									alignSelf: 'center',
									marginBottom: 20,
									marginTop: 10
								} }
							>
								No More Previous Messages
							</Text>
					}

					<View style={ styles.messagesContainer }>
						{
							this.props.messages.map((message: Message) => this._renderMessage(message))
						}
					</View>

				</ScrollView>

				<View style={ { bottom: 0, position: 'absolute', left: 0, right: 0, backgroundColor: 'white' } }>
					<TextInput
						style={ styles.messageInput }
						placeholder='Your message'
						onChangeText={ (messageText: string): void => this.setState({ messageText }) }
						value={ this.state.messageText }
						autoCompleteType='off'
						autoCorrect={ false }
					/>
					<TouchableOpacity
						style={ styles.sendButton }
						onPress={ this._sendMessage }
					>
						<Icon
							name='paper-plane'
							size={ 24 }
							solid
							style={ [ styles.sendButtonIcon, !this.state.messageText ? { color: '#555' } : { } ] }
						/>
					</TouchableOpacity>

					{
						!!this.state.newMessageCount &&
							<TouchableOpacity
								style={ {
									position: 'absolute',
									top: -60,
									width: width * 0.8,
									alignSelf: 'center'
								} }
								onPress={ this._scrollToBottom }
							>
								<Animated.View
									style={ {
										backgroundColor: 'orange',
										padding: 12,
										borderRadius: 8,
										shadowOffset: {
											width: 0,
											height: 4
										},
										shadowOpacity: 0.6,
										shadowRadius: 4,
										elevation: 12,
										opacity: this.animatedStyles.newMessagesButtonOpacity
									} }
								>
									<Text style={ { color: 'white', fontWeight: 'bold', alignSelf: 'center' } }>{ this.state.newMessageCount } New Messages</Text>
								</Animated.View>
							</TouchableOpacity>
					}
				</View>

				{
					this.state.showLoadingBar &&
						<Animated.View
							style={ {
								height: 5,
								width: this.animatedStyles.loadingBarWidth,
								backgroundColor: 'red',
								position: 'absolute',
								top: 0,
								left: 0
							} }
						/>
				}
			</KeyboardAvoidingView>
		);
	}
}

const mapStateToProps = (state: AppState): ChatState => ({
	...state.chatReducer,
	...state.chatMessagesReducer
});

export default connect(mapStateToProps, {
	chatSubscribe,
	chatUnsubscribe,
	getChatMessages
})(Chat);
