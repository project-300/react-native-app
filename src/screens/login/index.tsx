import React, { Component, ReactElement } from 'react';
import {
	Text,
	View,
	TextInput,
	TouchableOpacity
} from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';
import { Props, State } from './interfaces';
import { login } from '../../redux/actions';
import toastr from '../../helpers/toastr';
import { LoginState } from '../../types/redux-reducer-state-types';
import { AppState } from '../../store';
import Icon from 'react-native-vector-icons/FontAwesome5';

// Documentation: /docs/login.md

export class Login extends Component<Props, State> {

	public constructor(props: Props) {
		super(props);

		this.state = {
			username: '',
			password: '',
			hidePassword: true
		};
	}

	private _loginAttempt = async (): Promise<void> => {
		this.setState({ hidePassword: true });

		const { username, password } = this.state;

		if (!username) return toastr.error('Username is missing');
		if (!password) return toastr.error('Password is missing');

		const res: boolean = await this.props.login(username, password);
		res && this.props.navigation.navigate('Home');
	}

	public render(): ReactElement {
		return (
			<View style={ styles.container }>
				<View style={ styles.inputContainer }>
					<TextInput
						placeholder={ 'Username' }
						onChangeText={ (username: string): void => this.setState({ username }) }
						style={ styles.input } />
				</View>
				<View style={ styles.inputContainer }>
					<TextInput
						secureTextEntry={ this.state.hidePassword }
						placeholder={ 'Password' }
						style={ styles.input }
						onChangeText={ (password: string): void => this.setState({ password })}
					/>
					<TouchableOpacity
						style={ styles.showPasswordIconContainer }
						onPress={ (): void => this.setState({ hidePassword: !this.state.hidePassword }) }
					>
						<Icon
							name={ this.state.hidePassword ? 'eye' : 'eye-slash' }
							style={ styles.showPasswordIcon }
						/>
					</TouchableOpacity>
				</View>

				<TouchableOpacity
					disabled={ this.props.isLoggingIn }
					style={ styles.button }
					onPress={ this._loginAttempt }>
					<Text
						style={ styles.buttonText }
					>Login</Text>
				</TouchableOpacity>
				<TouchableOpacity
					onPress={ (): boolean => this.props.navigation.navigate('SignUp') }>
					<Text style={ styles.signUpLink }>
						Not registered yet? Sign Up
					</Text>
				</TouchableOpacity>
			</View>
		);
	}
}

const mapStateToProps = (state: AppState): LoginState => ({
	...state.loginReducer
});

export default connect(mapStateToProps, { login })(Login);
