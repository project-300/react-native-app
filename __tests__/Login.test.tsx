import React from 'react';
import { Login } from '../src/screens/login';
import renderer from 'react-test-renderer';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

jest.mock('react-native-aws3', () => { return { Request: { FormData: { } } }; });

const testProps = (props: { }): { } => ({
  navigation: {
	navigate: jest.fn()
  },
  ...props
});

configure({ adapter: new Adapter() });

describe('Login Screen', () => {
  // tslint:disable-next-line:no-any
  let props: any;
  let component: ShallowWrapper;
  let instance: React.Component<Login>;

  beforeEach(() => {
	props = testProps({ });
	component = shallow(<Login { ...props } />);
	instance = component.instance() as React.Component<Login>;
  });

  it('renders correctly', () => {
	expect(component.exists()).toBe(true);
  });

  it('matches snapshot', () => {
	const tree = renderer.create(<Login { ...props } />).toJSON();
	expect(tree).toMatchSnapshot();
  });
});
