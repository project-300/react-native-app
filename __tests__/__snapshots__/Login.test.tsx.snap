// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Username"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#DDD",
          "borderRadius": 4,
          "marginBottom": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#DDD",
          "borderRadius": 4,
          "marginBottom": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#CCC",
          "borderBottomRightRadius": 4,
          "borderTopRightRadius": 4,
          "height": 47.5,
          "marginRight": -10,
          "opacity": 1,
          "paddingTop": 16,
          "position": "absolute",
          "right": 10,
          "top": 0,
          "width": 40,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "black",
              "fontSize": 18,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        ÔÅÆ
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#02b2cc",
        "borderRadius": 4,
        "opacity": 1,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Login
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "marginTop": 40,
          "textDecorationLine": "underline",
        }
      }
    >
      Not registered yet? Sign Up
    </Text>
  </View>
</View>
`;
